tests/api/test_adgroup.py

```diff
@@ -0,0 +1,113 @@
+        assert_equal(response.status_code, 201)
+        response = self.client.post(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 400)
+
+    def test_post_suggested_tile(self):
+        url = url_for('api.adgroup.adgroups', campaign_id=self.new_campaign_id)
+        adgroup = dict(self.new_adgroup)
+        adgroup["type"] = "suggested"
+        adgroup["category"] = "Technology_General"
+        data = json.dumps(adgroup)
+        response = self.client.post(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 201)
+
+    def test_post_with_missing_category(self):
+        url = url_for('api.adgroup.adgroups', campaign_id=self.new_campaign_id)
+        invalid_adgroup = dict(self.new_adgroup)
+        invalid_adgroup["type"] = "suggested"
+        data = json.dumps(invalid_adgroup)
+        response = self.client.post(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 400)
+
+    def test_http_put(self):
+        url = url_for('api.adgroup.adgroups', campaign_id=self.new_campaign_id)
+        data = json.dumps(self.new_adgroup)
+        response = self.client.post(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 201)
+        new = json.loads(response.data)["message"]
+
+        url = url_for('api.adgroup.adgroup', campaign_id=self.new_campaign_id, adgroup_id=new["id"])
+        new["name"] = "new_name"
+        new["frequency_cap_daily"] = 5
+        new["frequency_cap_total"] = 20
+        data = json.dumps(new)
+        response = self.client.put(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 200)
+
+    def test_http_put_404(self):
+        url = url_for('api.adgroup.adgroups', campaign_id=self.new_campaign_id)
+        data = json.dumps(self.new_adgroup)
+        response = self.client.post(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 201)
+
+        url = url_for('api.adgroup.adgroup', campaign_id=1000, adgroup_id=1001)
+        response = self.client.put(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 404)
+
+    def test_http_put_400(self):
+        url = url_for('api.adgroup.adgroups', campaign_id=self.new_campaign_id)
+        data = json.dumps(self.new_adgroup)
+        response = self.client.post(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 201)
+        new_adgroup = json.loads(response.data)["message"]
+
+        url = url_for('api.adgroup.adgroup', campaign_id=self.new_campaign_id, adgroup_id=new_adgroup["id"])
+        new_adgroup["type"] = "suggested"
+        data = json.dumps(new_adgroup)
+        response = self.client.put(url, data=data, content_type="application/json")
+        assert_equal(response.status_code, 400)
```

Comment:
It isn't immediately clear to me why this is a 400. Maybe a doc string or comment would help.